### PR TITLE
Disable http/http2 timeout for sse events

### LINKED ISSUE #298: Option to disable request timeout
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Option to completely disable the request timeout (i.e. not pass anything in [`requestOptions.timeout`](https://github.com/fastify/fastify-reply-from/blob/master/lib/request.js#L268)).

### Motivation

We have an app that uses SSE, but it doesn't send requests directly to that server. It sends them to another server that uses `fastify-reply-from` to forward the request to the server that sends the events. Due to the persistent http connection, we'll always get a timeout error after whatever was configured for the `opts.http.requestOptions.timeout`.

We want to disable the timeout in this package because we still want to use this proxy requests to a from one server to another.

### Example

To avoid a breaking change, perhaps `timeout` could also be a `boolean`. So code that looks like below. Other options like allowing `null` to mean no timeout are there, but may break someone's code.

```js
// request.js line 267
if (
  // if timeout is not a boolean and falsy, use the default. 
  // if timeout is a boolean, use the default only if it's true
  typeof httpOpts.requestOptions.timeout !== 'boolean' && !httpOpts.requestOptions.timeout ||
  httpOpts.requestOptions.timeout
) {
  httpOpts.requestOptions.timeout = 10000
}
```
```js
// example usage
proxy.register(require('@fastify/reply-from'), {
  base: 'http://localhost:3001/'
  http: {
    requestOptions: {
      timeout: false
    }
  }
})
```

### PR DESCRIPTION
**Pull request by deanhaleem**
Related to #298. It was mentioned there `undici` possibly has a bug regarding this. I will work on creating a small reproduction and create an issue on that module

This checks the response headers for `content-type: text/event-stream`, and if found, get rid of any existing timeouts.

Fixes #298

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/http-timeout.test.js [hunk 1]
```diff
@@ -52,6 +52,31 @@ async function main () {
...
-main()
+test('http sse removes timeout test', async (t) => {
+  const target = Fastify()
+  t.teardown(target.close.bind(target))
+
+  target.get('/', (request, reply) => {
+    t.pass('request arrives')
+
+    reply.header('content-type', 'text/event-stream').status(200).send('hello world')
+  })
```
Comment (deanhaleem) [inline test/http-timeout.test.js:65]: I admit I was struggling a little figuring out the best way to test these changes. The way this works is it more tests the code via the coverage: without the new tests, the coverage lowers.

If it's preferred to instead specifically test that `req.setTimeout(0)` is being called, for sure can let me know

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by deanhaleem** [comment]
> Can you please open an issue on undici showing that the request is timing out if there is a SSE response? So we can track it upstream.
> 
> I think it would be better to have the http2 changes here.

Thanks, added the http2 changes back 👍  realized the initial reason it didn't work was missing the `http2Client` timeout.

And yes I will add an issue, need to find time to create a small reproduction first to make sure it's not an issue in my application.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**